kin-devplatform-sdk/src/main/java/kin/devplatform/base/KinEcosystemBaseActivity.java

```diff
@@ -0,0 +1,23 @@
+package kin.devplatform.base;
+
+import android.os.Bundle;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import kin.devplatform.KinEcosystemInitiator;
+import kin.devplatform.core.util.ExecutorsUtil;
+
+public abstract class KinEcosystemBaseActivity extends AppCompatActivity {
+
+	protected abstract @LayoutRes
+	int getLayoutRes();
+
+	@Override
+	protected void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(getLayoutRes());
+
+		KinEcosystemInitiator initiator = new KinEcosystemInitiator(new ExecutorsUtil());
```

Comment:
Why we do it for each activity?

Reply:
yep, there's a problem here, I made now `KinEcosystemInitiator` a singleton, so subsequent calls will do nothing if already init.